dotCMS/src/main/java/com/dotcms/management/filters/InfrastructureManagementFilter.java

```diff
@@ -0,0 +1,157 @@
+            try {
+                int proxyPort = Integer.parseInt(originalPort);
+                if (proxyPort == managementPort) {
+                    return true;
+                }
+            } catch (NumberFormatException e) {
+                // Invalid port header
+            }
+        }
+        
+        // Check if management port enforcement is disabled (for development/testing)
+        boolean strictChecking = Config.getBooleanProperty(InfrastructureConstants.Ports.STRICT_CHECK_PROPERTY, true);
+        if (!strictChecking) {
+            Logger.debug(this, "Management port strict checking disabled - allowing access");
+            return true;
+        }
+        
+        return false;
+    }
+
+    /**
+     * Gets the management port from the same environment variable used by server.xml.
+     * This ensures we use the exact same port configuration without duplication.
+     * 
+     * server.xml uses: port="${CMS_MANAGEMENT_PORT:-8090}"
+     * We check both the direct environment variable and Config system for testing.
+     */
+    private int getManagementPortFromEnvironment() {
+        // First check the exact same environment variable that server.xml uses
+        String portEnv = System.getenv(InfrastructureConstants.Ports.MANAGEMENT_PORT_PROPERTY);
+        if (portEnv != null && !portEnv.trim().isEmpty()) {
+            try {
+                return Integer.parseInt(portEnv.trim());
+            } catch (NumberFormatException e) {
+                Logger.warn(this, "Invalid CMS_MANAGEMENT_PORT environment variable: " + portEnv + 
+                          ", using default: " + InfrastructureConstants.Ports.DEFAULT_MANAGEMENT_PORT);
+            }
+        }
+        
+        // Fallback to Config system (handles DOT_ prefixed env vars and properties for testing)
+        try {
+            return Config.getIntProperty(InfrastructureConstants.Ports.MANAGEMENT_PORT_PROPERTY, 
+                                       InfrastructureConstants.Ports.DEFAULT_MANAGEMENT_PORT);
+        } catch (Exception e) {
+            // Handle ConversionException or other Config exceptions gracefully
+            Logger.warn(this, "Invalid CMS_MANAGEMENT_PORT configuration: " + e.getMessage() + 
+                      ", using default: " + InfrastructureConstants.Ports.DEFAULT_MANAGEMENT_PORT);
+            return InfrastructureConstants.Ports.DEFAULT_MANAGEMENT_PORT;
+        }
+    }
+
+    /**
+     * Sends a standardized access denied response for management endpoints.
+     */
+    private void sendManagementAccessDenied(HttpServletResponse response) throws IOException {
+        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        response.setContentType(InfrastructureConstants.Responses.CONTENT_TYPE_TEXT);
+        response.setCharacterEncoding(InfrastructureConstants.Responses.CHARSET_UTF8);
+        response.getWriter().write(InfrastructureConstants.Responses.ACCESS_DENIED_MESSAGE);
```

Comment:
Maybe we don't need this?  It becomes a dotCMS system identifier/tell - if you can request `/dotmgt` and see this message, you know it is running dotCMS.

I think a generic "not found" message might be better?

Reply:
I know what you mean looking further this is the message on the non-management port,  it would be better to just return a 404 probably